### ISSUE TITLE
Issue with Models not loading their Textures

### ISSUE THREAD

**Issue by SirCraftypants** [PluginBugs/Issues-ItemsAdder#2357]
### Terms

- [X] I'm using the very latest version of ItemsAdder and its dependencies.
- [X] I am sure this is a bug and it is not caused by a misconfiguration or by another plugin.
- [X] I already searched on this [Github page](https://github.com/PluginBugs/Issues-ItemsAdder/issues) to check if the same issue was already reported.
- [X] I already searched on the [plugin wiki](https://itemsadder.devs.beer/) to know if a solution is already known.
- [X] I already searched on the [forums](https://forum.devs.beer/) to check if anyone already has a solution for this.

### Discord tag (optional)

sircraftypants#4875

### What happened?

I have some custom models that use the entity warden texture, it was working fine in Itemsadder version 2 but, now that i've updated to version 3, it is refusing to allow the models to locate the warden texture located in

minecraft/textures/entity/warden/warden.png

### Steps to reproduce the issue

1. Create and install a model that uses the warden texture.
2. See that it loads the model but with the broken purple/black textures.

### Server version

[15:42:07 INFO]: This server is running Paper version git-Paper-392 (MC: 1.19.3) (Implementing API version 1.19.3-R0.1-SNAPSHOT) (Git: 0c9ace8)

### ItemsAdder Version

[17:13:50 INFO]: ItemsAdder version 3.3.1
[17:13:50 INFO]: Customized Minecraft without client mods!
[17:13:50 INFO]: Website: devs.beer
[17:13:50 INFO]: Author: LoneDev

### ProtocolLib Version

[17:15:21 INFO]: ProtocolLib version 5.0.0-SNAPSHOT-b612
[17:15:21 INFO]: Provides read/write access to the Minecraft protocol.
[17:15:21 INFO]: Authors: dmulloy2 and comphenix

### LoneLibs Version

[17:15:57 INFO]: LoneLibs version 1.0.23
[17:15:57 INFO]: Collection of libs used by my plugins
[17:15:57 INFO]: Authors: LoneDev and others

### Problematic items yml configuration file (optional)

```shell
items:
       #########SWORD##########
  warden_sword:
    display_name: '&aWarden Sword'
    resource:
      material: STONE_SWORD
      generate: false
      model_path: item/warden_sword
    durability:
      max_custom_durability: 2980
    attribute_modifiers:
      mainhand:
        attackDamage: 8.0
        attackSpeed: 2.0
```

**Comment by SirCraftypants** [comment]
This is how it looks in the models in question 

{
	"credit": "Made with Blockbench",
	"parent": "wooden_axe",
	"textures": {
		"4": "entity/warden/warden"
	},

**Comment by MrOmnes** [comment]
Same issue here

**Comment by LoneDev6** [comment]
Because it's how the game works.
Putting items textures inside folders which are not in `textures/item` or `textures/block` it's not working out of the box in 1.19.3 due to recent game changes.

ItemsAdder itself should fix this if you enable `1_19_3_atlas_support: true`, check if this option is enabled.

**Comment by SirCraftypants** [comment]
> Because it's how the game works. Putting items textures inside folders which are not in `textures/item` or `textures/block` is not working out of the box in 1.19.3 due to recent game changes.
> 
> ItemsAdder itself should fix this if you enable `1_19_3_atlas_support: true`, check if this option is enabled.

Yes it's set to true be default. 

That's the issue, the entity folder in the minecraft namespace isn't in `textures/item`, it's in textures/entity` if recent minecraft changes are the cause of if no longer working and the above atlas setting is there to fix it then it's not fixing it.

**Comment by LoneDev6** [comment]
Open your generated.zip (you must disable zip protection in config.yml, then regenerate the pack to be able to open it) and post here your `assets/minecraft/atlas/block.json` file

**Comment by SirCraftypants** [comment]
> Open your generated.zip (you must disable zip protection in config.yml, then regenerate the pack to be able to open it) and check if it contains `assets/minecraft/atlas/block.json` the warden path

Yes

`    {
      "type": "single",
      "resource": "entity/warden/warden"
    },`

**Comment by LoneDev6** [comment]
Is this texture actually available in the game itself?

Also, did you check your CLIENT log, as you always should, to check if there is any error about that?
https://itemsadder.devs.beer/faq/identify-why-textures-are-not-shown

**Comment by SirCraftypants** [comment]
> Is this texture actually available in the game itself?
> 
> Also, did you check your CLIENT log, as you always should, to check if there is any error about that? https://itemsadder.devs.beer/faq/identify-why-textures-are-not-shown

yes, it's minecrafts entity warden texture

I'm getting a warn for each item that uses the warden texture 

`[13:21:57] [Worker-Main-10/WARN]: Missing textures in model minecraft:shield#inventory:
    minecraft:textures/atlas/blocks.png:minecraft:entity/warden/warden`

But the texture exists in my resource pack and is inside the block.json too

**Comment by LoneDev6** [comment]
Works fine for me.
Probably the model failed to load for some reason, I don't know why you set the parent to `stone_sword`, if it's not a model created by you in this path for some reason `assets/minecraft/models/stone_sword.json` then this doesn't exists in the game.
Usually you set `item/generated`, `item/handheld` or no parent at all.
![image](https://user-images.githubusercontent.com/27242001/222136896-8e206cf9-ea2b-405f-a8c9-6a5378a1f538.png)
[test_issue_2357.zip](https://github.com/PluginBugs/Issues-ItemsAdder/files/10860847/test_issue_2357.zip)

**Comment by SirCraftypants** [comment]
the parent set you point to was only set for one of the models, there are 9 other models in the set that also don't load. Note i removed that parent as soon as i found it but it made not the slightest difference. This parent set has nothing to do with this problem.

The very same 10 models using the warden texture were working perfectly fine in Itemsadder version 2, all i did with update to itemsadder version [3.3.0b-r5] and everything was again fine, then i updated to 3.3.1 and it broke them.

**Comment by SirCraftypants** [comment]
Right I've tried 

`{
	"credit": "Made with Blockbench",
	"textures": {
		"4": "minecraft:entity/warden/warden"
	},`

and 

`{
	"credit": "Made with Blockbench",
	"textures": {
		"4": "entity/warden/warden"
	},`

The models were working perfectly fine before 3.3.1 update but to be sure it wasn't a model issue, i took fresh copies from my backup.

Models will not use the entity textures under the Minecraft namespace.

**Comment by LoneDev6** [comment]
Send the model on Discord via private message, I'll check it.
Anyway check my example model and let me know if it loads fine as shown in my screenshot.

**Comment by SirCraftypants** [comment]
> Send the model on Discord via private message, I'll check it. Anyway check my example model and let me know if it loads fine as shown in my screenshot.

IA's generating the folder `atlas` as `atlases`, i suspect this is the issue?

**Comment by LoneDev6** [comment]
It's correctly called `atlases`

**Comment by SirCraftypants** [comment]
I did a complete fresh install on Itemsadder 3.4.0c-beta

I added my Warden set, ran /iazip
I downloaded the generated.zip file and unzipped it
I checked the block.json file in atlases to confirm that the warden texture was present and is.

I also added some custom models to customize minecrafts tools. 

result:  Warden models still only load without it's warden texture and not even the custom models for minecrafts tools will load. still showing the default tools.

`[11:18:31] [Worker-Main-5/WARN]: Missing textures in model minecraft:stone_shovel#inventory:
    minecraft:textures/atlas/blocks.png:minecraft:entity/warden/warden`

I can confirm that `minecraft:entity/warden/warden` exists in block.json under atlases

I can also confirm that the texture is absolutely not missing, it's located in `assets/minecraft/textures/entity/warden/warden.png`

Despite the client claiming it doesn't exist.

I have checked the configurations a million times, you've already tested the shovel model yourself.

This has all been done on a test server with only Itemsadder and it's dependencies on Paper 1.19.3 latest as of today.

I even gave you a copy of the shovels configuration and the model.

So my question is, why is it working for you but not for me and why are the custom models i added for the minecraft tools in assets/minecraft/models/item. They are all named exactly the same as the default tool json's. 

Now all this was perfectly fine pre 3.3.1 and nothing changed between then and now, nothing what so ever as i haven''t worked on my server at all. All i've been doing is regularly updating.
